src/extractor/extraction_containers.cpp

```diff
@@ -453,11 +453,13 @@ void ExtractionContainers::PrepareEdges(lua_State *segment_state)
                 all_edges_list[i].result.weight < min_forward_weight)
             {
                 min_forward_idx = i;
+                min_forward_weight = all_edges_list[i].result.weight;
```

Comment:
Can we add a test-case with multiple parallel speed-edges and empty name and check for speed?